src/mjolnir/landmarks.cc

```diff
@@ -296,6 +318,226 @@ bool BuildLandmarkFromPBF(const boost::property_tree::ptree& pt,
+                                                // stations that are off the road a bit (for parking)
+
+        // call loki::Search to get nearby edges to each landmark
+        std::unordered_map<valhalla::baldr::Location, PathLocation> result =
+            loki::Search({landmark_location}, reader, sif::CreateNoCost({}));
+        if (result.size() > 1) {
+          throw std::logic_error("Error occurred in calling loki::Search. Result size is " +
+                                 std::to_string(result.size()) + ", but should be one or zero");
+        }
+        if (result.size() == 0) {
+          continue;
+        }
+
+        std::vector<PathLocation::PathEdge> edges = result.begin()->second.edges;
+        // for each edge insert edgeid - landmark_pkey pair into the sequence file
+        // TODO: maybe do some filtering and only keep some of the edges it finds?
+        for (const auto& edge : edges) {
+          seq_file.push_back(std::make_pair(edge.id, landmark.id));
+        }
+      }
+    }
+  }
+  // sort the sequence by tiles
+  seq_file.sort(sort_seq_file);
+
+  seq_file_name.set_value(file_name);
+}
+
+// Update tiles to associate landmarks, return some stats about the numbers of updated tiles, edges,
+// and landmarks
+void UpdateTiles(midgard::sequence<std::pair<GraphId, uint64_t>>& seq_file,
+                 const std::string& tile_dir,
+                 const std::string& db_name,
+                 const size_t& thread_number,
+                 const size_t& total_threads,
+                 std::promise<std::tuple<size_t, size_t, size_t>>& stats) {
+  // open the database and initialize a unique pointer to graph tile builder
+  std::unique_ptr<GraphTileBuilder> tile_builder_ptr = nullptr;
+
+  // const std::string dbname = pt.get_child("mjolnir").get<std::string>("landmarks", "");
+  LandmarkDatabase db(db_name, true);
+
+  // stats
+  size_t updated_tiles = 0, updated_edges = 0, updated_landmarks = 0;
+  GraphId last_edge, last_tile;
+
+  // modulus count to assign tiles to threads
+  size_t tile_count = static_cast<size_t>(-1);
+
+  for (auto it = seq_file.begin(); it != seq_file.end(); ++it) {
+    if ((*it).first.Tile_Base() != last_tile) {
+      last_tile = (*it).first.Tile_Base();
+      tile_count++;
+    }
+    if (tile_count % total_threads != thread_number) {
+      continue;
+    }
+
+    if (!tile_builder_ptr ||
+        tile_builder_ptr->header_builder().graphid().Tile_Base() != (*it).first.Tile_Base()) {
+      // store the previously updated tile
+      if (tile_builder_ptr) {
+        tile_builder_ptr->StoreTileData();
+        updated_tiles++;
+      }
+      // reset the tile builder ptr to the current tile
+      tile_builder_ptr.reset(new GraphTileBuilder(tile_dir, (*it).first.Tile_Base(), true));
+    }
+
+    // retrieve the landmark to be added
+    const std::vector<Landmark> landmark =
+        db.get_landmarks_by_ids({static_cast<int64_t>((*it).second)});
```

Comment:
in the future we can do batches of ids, though it will complicate the code it will likely speed up the processing